File: 1389.按既定顺序创建目标数组/1389.按既定顺序创建目标数组.py
'''
Descroption: LeetCode 1387. 将整数按权重排序
Author: EmoryHuang
Date: 2021-08-27 15:22:49
Method:
直接 insert
'''


class Solution:
    def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:
        ans = []
        for i in range(len(nums)):
            ans.insert(index[i], nums[i])
        return ans

File: Solution/1389.按既定顺序创建目标数组/1389.按既定顺序创建目标数组.cpp
/*
 * @Descroption: LeetCode 1387. 将整数按权重排序
 * @Author: EmoryHuang
 * @Date: 2021-08-27 15:23:58
 * @Method:
 * 直接 insert
 */

class Solution {
   public:
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
        vector<int> ans;
        for (int i = 0; i < nums.size(); i++) ans.insert(ans.begin() + index[i], nums[i]);
        return ans;
    }
};